### PR TITLE
Updating NFS ganesha chapter in admin and deploy

### LINKED ISSUE #370: Suspicious Admin Guide sections (mainly from DeepSea perspective)
This issue lists sections which are obviously wrong and need to be either removed from the guide or updated to match SES7:

- [x] --- Admin Tasks ---
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ds-osd-recover
- manually deploy OSDs: ceph-volume to create devices, deploy OSDs from containers, we need a seprate JIRA issue here https://jira.suse.com/browse/SES-2110
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#moving-saltmaster
- deploy new node + restore salt-master + ceph-salt import  from exported backup
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#deepsea-rolling-updates
- underlying OS: ceph-salt update (https://github.com/ceph/ceph-salt/pull/303)
- containerized services: cephadm (https://docs.ceph.com/docs/master/cephadm/upgrade/)
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#sec-salt-cluster-reboot
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#admin-apparmor
- remove the whole section
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#deepsea-ceph-purge
- ceph-salt purge

- [x] --- Cluster Backup ---
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#sec-deployment-backup-deepsea
- remove
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#backup-config-files
- no openattic, no ceph.conf
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#restore-ceph
- remove step 4, 9-end

- [ ] --- Monitoring & Alerting ---
- [ ] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#monitoring-alerting
- talk to Patrick Seidensal @p-se 
- not as flexible as in SES7, config options must be passed to cephadm command, it re-generates their config files
- https://jira.suse.com/browse/SES-1194
- move '7.5 Troubleshooting Alerts' to Troubleshooting guide
- replace systemd units with `ceph orch demon restart osd.123`

- [x] --- RGW ---
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ogw-deploy
- YAML based import
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-rgw-operating
- dont use systemd unit files
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ogw-selfcert (is there a common SSL section?)
- remove DS specific stuff + https://confluence.suse.com/pages/viewpage.action?pageId=505119148 has most of the info
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ogw-ssl-simple, https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ogw-ssl-advanced
- not needed, ceph-salt takes care
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-rgw-sync-zones-configure (bullet 5)
- dont use systemd unit files but `ceph orch restart rgw.realm.zone`
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-rgw-fed-masterzone-startrgw
- dont use systemd unit files but `ceph orch restart rgw.realm.zone`
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-rgw-fed-secondzone-startrgw
- dont use systemd unit files but `ceph orch restart rgw.realm.zone`
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-rgw-failover-dr (bullets 3,4,7)
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-rgw-single-site-multi (bullet 7)
- move to the upgrade process
- [x] 12.13.3.3 Update the Ceph Configuration File
- remove the section
- [x] 6.5.3.4 Deploy Object Gateways
- steps 1-3 are optional, if realm, zone do not exist, they are created 

12.14 Load Balancing the Object Gateway Servers with HAProxy is going to be handled by orchestrator automatically

- [ ] --- NFS Ganesha ---
- [ ] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-nfsganesha-config
- Bug filed to ensure we have man page access #SES-2108
- [ ] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-nfsganesha-config-service-general
- @mgfritch  required here. The files are wrong, and the config files are actually stored in the middle layer. See: https://confluence.suse.com/display/SUSEEnterpriseStorage/nfs-ganesha
- [ ] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-nfsganesha-customrole-rgw-multiusers
- @mgfritch  to review. Config files are all different now. https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/templates/services/nfs/ganesha.conf.j2
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-nfsganesha-customrole-rgw-cephfs
- Remove chapter completely.
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-nfsganesha-services
- Use the `ceph orch restart nfs.<service.id>` command instead

- [x] --- Ceph Point Releases ---
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#id-1.4.9

### PR DESCRIPTION
**Pull request by asettle**
Partial-fix #370

### CODE REVIEW

File: xml/deploy_cephadm.xml [hunk 1]
```diff
@@ -2157,48 +2157,21 @@ placement:
...
-      </para>
-     </listitem>
-     <listitem>
-      <para>
-       <replaceable>EXAMPLE_POOL</replaceable> with the name of the pool where
-       the &ganesha; RADOS configuration object will be stored.
-      </para>
-     </listitem>
-     <listitem>
+<screen>&prompt.cephuser;ceph orch apply nfs foo nfs-ganesha nfs-ns</screen>
```
Comment (lproven) [inline xml/deploy_cephadm.xml:2168]: I suggest a [replaceable] tag here.

Comment (asettle) [inline xml/deploy_cephadm.xml:2168]:
```suggestion
<screen>&prompt.cephuser;ceph orch apply nfs <replaceable>foo</replaceable> nfs-ganesha nfs-ns</screen>

**Review by tbazant — APPROVED** [review]
GOOD

File: xml/admin_nfsganesha.xml [hunk 2]
```diff
@@ -51,9 +47,29 @@
   <title>Installation</title>
 
   <para>
-   For installation instructions, see <!-- DEAD_LINK -->.
+   For installation instructions, see <xref linkend ="deploy-cephadm-day2-service-nfs"/>.
   </para>
  </sect1>
+
+ <sect1 xml:id="ceph-nfsganesha-namespace">
+   <title>Namespace Conventions</title>
```
Comment (mgfritch) [inline xml/admin_nfsganesha.xml:55]:
```suggestion
   <title>&ogw Namespace Conventions</title>
```

we should probably be careful to distinguish :
* RADOS pool namespace
* RGW namespace

File: xml/admin_nfsganesha.xml [hunk 3]
```diff
@@ -51,9 +47,29 @@
...
+   <para>
+     The top level of the attached namespace consists of S3 buckets,
+     represented as NFS directories. Files and directories subordinate to
+     buckets are each represented as objects, following S3 prefix and delimiter
+     conventions, with <literal>/</literal> being the only supported path delimiter.
+   </para>
+   <para>
+    S3 buckets are exported to NFS on a per-user basis, via this path convention:
+    <filename><replaceable>GANESHA_NODE:</replaceable>/<replaceable>USERNAME</replaceable>/<replaceable>BUCKETNAME</replaceable></filename>.
+   </para>
```
Comment (mgfritch) [inline xml/admin_nfsganesha.xml:70]:
```suggestion
```
Can also probably be removed .. S3 buckets are not exported by default and they are mounted using the explicitly defined pseudo path.

```
$ mount GANESHA_NODE:/pseudo-path /mnt
```

File: xml/admin_nfsganesha.xml [hunk 4]
```diff
@@ -745,17 +665,16 @@ ganesha_configurations:
...
-&prompt.sminion;<command>systemctl</command> start nfs-ganesha</screen>
+<screen>&prompt.cephuser;ceph orch start nfs.<replaceable>SERVICE_ID</replaceable></screen>
 
   <para>
-   Restart &ganesha; with:
+   To restart the &ganesha; service, run:
   </para>
 
-<screen>&prompt.sminion;<command>systemctl</command> restart nfs-ganesha</screen>
+<screen>&prompt.cephuser;ceph orch restart nfs.<replaceable>SERVICE_ID</replaceable></screen>
```
Comment (mgfritch) [inline xml/admin_nfsganesha.xml:677]: The `systemctl` examples would start/restart a single NFS ganesha daemon, but `ceph orch restart..` will restart the entire NFS service (and all the containing daemons)... 

should we use `ceph orch daemon ...` instead? 

or maybe reference the generic section on this topic? 
https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#cha-ceph-operating-individual

File: xml/admin_nfsganesha.xml [hunk 5]
```diff
@@ -30,10 +30,6 @@
   own storage mechanism&mdash;such as &ceph;&mdash;and access it from any NFS
   client.
  </para>
- <para>
-  S3 buckets are exported to NFS on a per-user basis, for example via the path
-  <filename><replaceable>GANESHA_NODE:</replaceable>/<replaceable>USERNAME</replaceable>/<replaceable>BUCKETNAME</replaceable></filename>.
- </para>
  <para>
   A &cephfs; is exported by default via the path
   <filename><replaceable>GANESHA_NODE:</replaceable>/cephfs</filename>.
```
[the remark this answers is not in the document]
Comment (bk201) [reply xml/admin_nfsganesha.xml:35]: +1 for removing.

**Comment by bk201** [comment]
Near the end of [16.2 Configuration](https://susedoc.github.io/doc-ses/master/html/ses-admin/cha-ceph-nfsganesha.html#ceph-nfsganesha-config) (`Note that changes to this file are overwritten when DeepSea stage 4 ...`) is DeepSea-only

[16.2.1 Service Configuration](https://susedoc.github.io/doc-ses/master/html/ses-admin/cha-ceph-nfsganesha.html#ceph-nfsganesha-config-service-general) is DeepSea-only.

[16.3.1 Different Object Gateway Users for NFS Ganesha](https://susedoc.github.io/doc-ses/master/html/ses-admin/cha-ceph-nfsganesha.html#ceph-nfsganesha-customrole-rgw-cephfs) is also DeepSea-only and need to be removed.
This makes the entire section `16.3 Custom NFS Ganesha Roles` be empty.

File: xml/deploy_cephadm.xml [hunk 6]
```diff
@@ -2188,17 +2184,13 @@ spec:
...
-       information on NFS namespaces, see
-       <link xlink:href="https://docs.ceph.com/docs/master/radosgw/nfs/#namespace-conventions
-"/>
-<!-- 2020-06-10 tbazant: possibly import that upstream section into
-       Admin Guide? -->
-      </para>
+       <para>
+         <replaceable>EXAMPLE_NAMESPACE</replaceable> (optional) with the desired
+         NFS namespace (for example, <literal>nfs</literal>). For more
+         information on NFS namespaces, see <xref linkend="ceph-nfsganesha-namespace"/>.
```
Comment (bk201) [inline xml/deploy_cephadm.xml:2190]: Just like the previous comment from @mgfritch, the namespace here is a rados namespace, not a RGW namespace. So the reference is not appropriate.

**Review by tbazant — APPROVED** [review]
LGTM

**Pull request merged**